core/domain/base_model_validators_test.py

```diff
@@ -372,3 +388,105 @@ def test_error_raised_when_commit_message_large(self):
+            commit_type='create',
+            commit_cmds=[commit_cmd],
+            post_commit_status='public')
+        model.update_timestamps()
+        mock_validator = MockCommitLogEntryModelValidator()
+        mock_validator.errors.clear()
+        mock_validator.validate(model)
+        self.assertEqual(mock_validator.errors, {})
+
+    def test_error_raised_when_commit_cmd_invalid(self):
+        commit_cmd = {
+            'cmd': 'invalid_cmd'
+        }
+        model = MockCommitLogEntryModel(
+            id='mock-12345',
+            user_id='user_id',
+            commit_cmds=[commit_cmd])
+        model.update_timestamps()
+        mock_validator = MockCommitLogEntryModelValidator()
+        mock_validator.errors.clear()
+        mock_validator.validate(model)
+        self.assertDictContainsSubset(
+            {
+                'commit cmd invalid_cmd check': [
+                    'Entity id mock-12345: '
+                    'Commit command domain validation for command:'
+                    ' %s failed with error:'
+                    ' Command invalid_cmd is not allowed' % commit_cmd
+                ]
+            },
+            mock_validator.errors
+        )
+
+    def test_no_error_is_raised_when_commit_cmd_deprecated(self):
+        commit_cmd = {
+            'cmd': 'deprecated_command'
+        }
+        model = MockCommitLogEntryModel(
+            id='mock-12345',
+            user_id=self.owner_id,
+            commit_type='create',
+            commit_cmds=[commit_cmd],
+            post_commit_status='public')
+        model.update_timestamps()
+        mock_validator = MockCommitLogEntryModelValidator()
+        mock_validator.errors.clear()
+        mock_validator.validate(model)
+        self.assertEqual(mock_validator.errors, {})
+
+    def test_no_error_is_raised_when_commit_cmd_value_deprecated(self):
+        commit_cmd = {
+            'cmd': 'mock_command',
+            'mock_property': 'deprecated_value'
+        }
+        model = MockCommitLogEntryModel(
+            id='mock-12345',
+            user_id=self.owner_id,
+            commit_type='create',
+            commit_cmds=[commit_cmd],
+            post_commit_status='public')
+        model.update_timestamps()
+        mock_validator = MockCommitLogEntryModelValidator()
+        mock_validator.errors.clear()
+        mock_validator.validate(model)
+        self.assertEqual(mock_validator.errors, {})
+
+    def test_error_is_raised_when_commit_cmd_value_invalid(self):
+        commit_cmd = {
+            'cmd': 'mock_command',
+            'mock_property': 'invalid_value'
+        }
+        model = MockCommitLogEntryModel(
+            id='mock-12345',
+            user_id=self.owner_id,
+            commit_type='create',
+            commit_cmds=[commit_cmd],
+            post_commit_status='public')
+        model.update_timestamps()
+        mock_validator = MockCommitLogEntryModelValidator()
+        mock_validator.errors.clear()
```

Comment:
Do we really need to clear from a freshly-initialized object? If so, let's move that into the constructor/`__init__` function instead and remove this code here.

Reply:
I have removed the clear since it was not necessary. Done.